src/plugins/data/server/search/strategies/sql_search/sql_search_strategy.test.ts

```diff
@@ -0,0 +1,263 @@
+import { SqlSearchStrategyRequest } from '../../../../common';
+
+const mockSqlResponse = {
+  body: {
+    id: 'foo',
+    is_partial: false,
+    is_running: false,
+    rows: [],
+  },
+};
+
+describe('SQL search strategy', () => {
+  const mockSqlGetAsync = jest.fn();
+  const mockSqlQuery = jest.fn();
+  const mockSqlDelete = jest.fn();
+  const mockLogger: any = {
+    debug: () => {},
+  };
+  const mockDeps = {
+    esClient: {
+      asCurrentUser: {
+        sql: {
+          getAsync: mockSqlGetAsync,
+          query: mockSqlQuery,
+          deleteAsync: mockSqlDelete,
+        },
+      },
+    },
+    searchSessionsClient: createSearchSessionsClientMock(),
+  } as unknown as SearchStrategyDependencies;
+
+  beforeEach(() => {
+    mockSqlGetAsync.mockClear();
+    mockSqlQuery.mockClear();
+    mockSqlDelete.mockClear();
+  });
+
+  it('returns a strategy with `search and `cancel`, `extend`', async () => {
+    const esSearch = await sqlSearchStrategyProvider(mockLogger);
+
+    expect(typeof esSearch.search).toBe('function');
+    expect(typeof esSearch.cancel).toBe('function');
+    expect(typeof esSearch.extend).toBe('function');
+  });
+
+  describe('search', () => {
+    describe('no sessionId', () => {
+      it('makes a POST request with params when no ID provided', async () => {
+        mockSqlQuery.mockResolvedValueOnce(mockSqlResponse);
+
+        const params: SqlSearchStrategyRequest['params'] = {
+          query:
+            'SELECT customer_first_name FROM kibana_sample_data_ecommerce ORDER BY order_date DESC',
+        };
+        const esSearch = await sqlSearchStrategyProvider(mockLogger);
+
+        await esSearch.search({ params }, {}, mockDeps).toPromise();
+
+        expect(mockSqlQuery).toBeCalled();
+        const request = mockSqlQuery.mock.calls[0][0];
+        expect(request.query).toEqual(params.query);
+        expect(request).toHaveProperty('format', 'json');
+        expect(request).toHaveProperty('keep_alive', '1m');
+        expect(request).toHaveProperty('wait_for_completion_timeout');
+      });
+
+      it('makes a GET request to async search with ID', async () => {
+        mockSqlGetAsync.mockResolvedValueOnce(mockSqlResponse);
+
+        const params: SqlSearchStrategyRequest['params'] = {
+          query:
+            'SELECT customer_first_name FROM kibana_sample_data_ecommerce ORDER BY order_date DESC',
+        };
+
+        const esSearch = await sqlSearchStrategyProvider(mockLogger);
+
+        await esSearch.search({ id: 'foo', params }, {}, mockDeps).toPromise();
+
+        expect(mockSqlGetAsync).toBeCalled();
+        const request = mockSqlGetAsync.mock.calls[0][0];
+        expect(request.id).toEqual('foo');
+        expect(request).toHaveProperty('wait_for_completion_timeout');
+        expect(request).toHaveProperty('keep_alive', '1m');
+        expect(request).toHaveProperty('format', 'json');
+      });
+    });
+
+    describe.skip('with sessionId', () => {
```

Comment:
Could you add a comment here linking to the same issue as before explaining why this is skipped

Reply:
Added 👍